mongoengine/queryset/transform.py

```diff
@@ -98,6 +99,15 @@ def query(_doc_cls=None, **kwargs):
                         value = value['_id']
 
             elif op in ('in', 'nin', 'all', 'near') and not isinstance(value, dict):
+                if isinstance(value, BaseDocument):
```

Comment:
Ahh, I missed that, thanks for clearing that up! I think in such case we can change this condition to:

```
# Raise an error if the in/nin/all/near param is not iterable. We need a
# special check for BaseDocument, because - although it's iterable - using
# it as such in the context of this method is most definitely a mistake.
if not hasattr(value, '__iter__') or isinstance(value, BaseDocument):
   raise TypeError(...)
```

The comment is crucial, too, given that this is a pretty unintuitive `isinstance` check.What do you think @malthejorgensen ?